figures/scripts/supp-S3-panels-BCD.R

```diff
@@ -0,0 +1,321 @@
+chr_end <- chr_start + summary(chrs)
+mid_points <- floor((chr_start + chr_end) / 2)
+
+# Make text labels for chromosome text
+chr_text <- ComplexHeatmap::anno_mark(
+  at = mid_points,
+  labels = levels(chrs),
+  which = "column",
+  side = "bottom",
+  labels_rot = 45,
+  labels_gp = grid::gpar(cex = 0.65)
+)
+
+# Create the Heatmap annotation object
+chr_annot <- HeatmapAnnotation(
+  df = data.frame(chrs),
+  col = list(chrs = chr_colors),
+  name = "",
+  show_legend = FALSE,
+  show_annotation_name = FALSE,
+  mark = chr_text, # Put the text in
+  border = TRUE
+)
+
+
+
+### Make row annotation object
+# Get the histologies for the samples in this set and order them by histology
+
+# Figure out cancer group counts, where NA display groups are _excluded_
+cancer_group_counts <- metadata %>%
+  drop_na(cancer_group_display) %>%
+  # No other ? TODO!
+  #filter(cancer_group_display != "Other") %>%
+  # IDs of interest 
+  filter(Kids_First_Biospecimen_ID %in% sample_ids) %>%
+  # count and create new label. we run factor code to ensure the right order
+  count(cancer_group_display, name = "cancer_group_n") %>%
+  mutate(
+    cancer_group_display = fct_reorder(cancer_group_display, cancer_group_n, .desc=T), 
+    # and now make the label
+    cancer_group_display_n = glue::glue("{cancer_group_display} (N = {cancer_group_n})")
+  ) %>%
+  # arrange so we're in the right order
+  arrange(cancer_group_display)
+
+samples_for_heatmap <- metadata %>%
+  # again, filter to IDs of interest
+  filter(Kids_First_Biospecimen_ID %in% sample_ids) %>%
+  select(Kids_First_Biospecimen_ID, 
+         cancer_group_display, # display names needed for factoring
+         cancer_group_hex) %>%
+  inner_join(
+    select(
+      cancer_group_counts,
+      cancer_group_n,
+      cancer_group_display
+    )
+  ) %>%
+  mutate(
+    cancer_group_display = fct_reorder(cancer_group_display, cancer_group_n, .desc=T), 
+    # and now relabel
+    cancer_group_display = factor(cancer_group_display, labels = cancer_group_counts$cancer_group_display_n)
+  ) %>%
+  # Make sure the rows are also in cancer_group_order
+  arrange(cancer_group_display) %>%
+  # Store as rownames
+  column_to_rownames("Kids_First_Biospecimen_ID") %>%
+  # remove coumn
+  select(-cancer_group_n)
+ 
+# Make a color key that's formatted for ComplexHeatmap
+# Get a distinct version of the color keys
+cancer_color_key_df <- samples_for_heatmap %>%
+  dplyr::select(cancer_group_display, cancer_group_hex) %>%
+  dplyr::distinct()
+
+# Make color key specific to these samples
+cancer_color_key <- cancer_color_key_df$cancer_group_hex
+names(cancer_color_key) <- cancer_color_key_df$cancer_group_display
+
+# Get coordinate start positions
+hist_start <- match(names(cancer_color_key), samples_for_heatmap$cancer_group_display)
+
+# Get coordinate end positions for each histology group
+hist_end <- hist_start + summary(samples_for_heatmap$cancer_group_display)
+
+# Get mid points of 
```

Comment:
Looks like this was incomplete in the original notebook, but taking my cue from the chromosome midpoints I think this is right...
```suggestion
# Get mid points of histology group for labels
```